com.unity.ml-agents/Runtime/Agent.cs

```diff
@@ -493,9 +493,10 @@ void SendInfoToBrain()
 
             m_Brain.RequestDecision(m_Info, sensors);
 
-            if (m_Recorder != null && m_Recorder.record && Application.isEditor)
+            // TODO simplify, only check if m_Recorder.GetExperienceWriter() exists
+            if (m_Recorder != null && m_Recorder.record)
             {
-                m_Recorder.WriteExperience(m_Info, sensors);
+                WriteExperience(m_Recorder.GetExperienceWriter());
```

Comment:
Thinking this might be better as a delegate instead; users could register their own experience writing methods.

Reply:
How do we anticipate our users customizing experience writing methods? We need to ensure that the flexibility we provide here is mirrored on the reader side (in Python).

Reply:
Obsolete - users just need to pass a Stream to the DemonstrationStore if they want to handle custom write logic.